gcp_variant_transforms/libs/vcf_header_parser.py

```diff
@@ -16,32 +16,23 @@
 
 from __future__ import absolute_import
 
-from collections import namedtuple
-from apache_beam.io.filesystems import FileSystems
-
 import vcf
 
-__all__ = ['HeaderFields', 'get_vcf_headers']
-
-
-# Stores parsed header information.
-#   - infos: Stores all `##INFO` header lines. Type is a ``dict`` with values
-#     equal to ``namedtuple`` defined in ``vcf.parser._Info``.
-#   - formats: Stores all `##FORMAT` header lines. Type is a ``dict`` with
-#     values equal to ``namedtuple`` defined in ``vcf.parser._Formats``.
-HeaderFields = namedtuple('HeaderFields', ['infos', 'formats'])
```

Comment:
I am thinking that VcfHeader actually belongs to this file because (i) all that this module currently has, in some sense, is a factory for VcfHeader objects (i.e., get_vcf_headers); and (ii) the dependency of this module on something from beam_io seems a little odd and I think a more logical dependency order is to have VcfHeader defined here and beam_io.vcf_header_io depend on this module instead (so the opposite direction).

If you make that change, many of the other dependencies in libs/* modules on beam_io will also be gone (specially those that this PR has introduced). So in general the dependency chain is beam_io modules depending on libs, not the opposite. WDYT?

Reply:
I actually think I/O modules should have the type in themselves (e.g. VCF I/O has Variant records). In any case, we're actually planning to remove VcfHeader altogether in favor of Nucleus [protos](https://github.com/google/nucleus/blob/master/nucleus/protos/variants.proto#L161), so it doesn't really matter at this point :)

Reply:
Well, you can argue that this module (`vcf_header_parser`) is actually an I/O module as well. In fact, both this module and the beam module use `vcf` which is the real I/O module IMO.

As a general guideline, it is important to pay attention to the "flow" of dependencies between packages and I have found that having Beam/Flume related functionality depend on more pure libraries is the right direction. Anyways, up to you, specially if you feel that the Nucleus change will happen soon, I am okay either way.